flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java

```diff
@@ -548,9 +549,57 @@ public ServerMapping getMappingById(@PathVariable("id") String id) {
    @RequestMapping(value = "/mapping/hostnameip", method = RequestMethod.POST)
    public void createHostNameIPMapping(@RequestBody AssetIPMapping mapping) {
       BaseDocumentUtil.generateID(mapping);
+      String ip = mapping.getIp();
+      if(ip == null || ip.isEmpty() || !ip.matches(HandleURL.IP_REGX)) {
+         throw new WormholeRequestException(HttpStatus.INTERNAL_SERVER_ERROR, "Invalid ip address",
+               null);
+      }
+      if (!assetService.isAssetNameValidate(mapping)) {
+         throw new WormholeRequestException(HttpStatus.INTERNAL_SERVER_ERROR, "Invalid asset name",
+               null);
+      }
       assetIPMappingRepository.save(mapping);
    }
 
+   @ResponseStatus(HttpStatus.OK)
+   @RequestMapping(value = "/mapping/hostnameip", method = RequestMethod.PUT)
+   public void updateHostNameIPMapping(@RequestBody AssetIPMapping mapping) {
+      BaseDocumentUtil.generateID(mapping);
+      AssetIPMapping oldMapping = assetIPMappingRepository.findOne(mapping.getId());
+      if(!assetService.isAssetNameValidate(mapping)) {
+         throw new WormholeRequestException(HttpStatus.INTERNAL_SERVER_ERROR,"Invalid asset name",null);
+      }
+      oldMapping.setAssetname(mapping.getAssetname());
+      assetIPMappingRepository.save(oldMapping);
+   }
+
+   @ResponseStatus(HttpStatus.OK)
+   @RequestMapping(value = "/mapping/hostnameip", method = RequestMethod.GET)
+   public Page<AssetIPMapping> getHostNameByRange(@RequestParam(value = "pagesize",required = false, defaultValue = "20") Integer pageSize,
+         @RequestParam(value = "pagenumber",required = false, defaultValue = "1") Integer pageNumber) {
+      if (pageNumber < FlowgateConstant.defaultPageNumber) {
+         pageNumber = FlowgateConstant.defaultPageNumber;
+      } else if (pageSize <= 0) {
+         pageSize = FlowgateConstant.defaultPageSize;
+      } else if (pageSize > FlowgateConstant.maxPageSize) {
+         pageSize = FlowgateConstant.maxPageSize;
+      }
+      PageRequest pageRequest = new PageRequest(pageNumber - 1, pageSize);
```

Comment:
Let's create a job to search mapping by IP , then we should allow user to edit the IPMapping base on the search result.